pulp_container/app/models.py

```diff
@@ -154,40 +164,93 @@ def init_image_nature(self):
             return self.init_manifest_nature()
 
     def init_manifest_list_nature(self):
+        updated_type = False
+        if not self.type:
+            self.type = MANIFEST_TYPE.UNKNOWN
+            updated_type = True
+
         for manifest in self.listed_manifests.all():
             # it suffices just to have a single manifest of a specific nature;
             # there is no case where the manifest is both bootable and flatpak-based
-            if manifest.is_bootable:
+            if manifest.type == MANIFEST_TYPE.BOOTABLE:
+                self.type = MANIFEST_TYPE.BOOTABLE
                 self.is_bootable = True
                 return True
-            elif manifest.is_flatpak:
+            elif manifest.type == MANIFEST_TYPE.FLATPAK:
+                self.type = MANIFEST_TYPE.FLATPAK
                 self.is_flatpak = True
                 return True
```

Comment:
Can a manifest list be of type "helm"?

Reply:
Manifest list should stay as `list`. Nowadays I've seen lists containing artifacts and images, so it's a mixture of things..

Reply:
Aah, good to know! I initially thought that if any of the listed manifests is of type "bla", then, we can treat the entire manifest list as being of type "bla". My goal was to reduce the number of API calls for our integrators, so they would not need to check each listed manifest individually to determine the type. By treating the manifest list as the "list" type, they would need to iterate through all listed manifests to get the right type and potentially present it in the UI (because users should see only the tagged manifests/manifest lists in the UI).

It might even make sense to allow filtering manifests by their referencing manifest list. Meaning that only one additional API call will be needed at most.

Reply:
Also, when it comes to different types of listed manifests, we might even consider prioritizing one over the other. The type that has the highest priority will be displayed in the view.

Reply:
So if we decide to set manifestlist as `type: list`, Pulp integrators would need to check and identify the corresponding manifests types by themselves. But if we set the type based on a single manifest from the list (for example, `flatpak`) there is a risk of providing the wrong type since we can now have a mixture of things. :thinking:

Reply:
If we decide to take a middle ground, it would be best to "allow filtering manifests by their referencing manifest list". Another PR. Also, note that this "type" field is pretty flexible. We can add or remove new types later.

Reply:
Unless, we do not enforce these types on the DB level...

Reply:
Initially manifest list was introduced for the multi-arch builds of the same image, nowadays it can also be treated as a list of images collected together. there is no enforcement in the specs around this.
Here's a practical example `skopeo inspect --raw docker://quay.io/nalind/testing:podman-machine-unified|jq`

I would not do any heuristics and just leave type as `list`. Integrators can take their own risks and parse and prioritize the types the consider more important on their own.

Reply:
@ianballou FYI the rationale behind ^

Reply:
@lubosmj  you should have inspected each child manifest and you would have seen that 2 of them are artifacts and 2 are different arches for a regular image.